test/unit/config/test__base.py

```diff
@@ -61,15 +65,15 @@ def test_base_config_init_with_ident(tmp_path):
     assert isinstance(rc.loader, ArtifactLoader)
 
 
-def test_base_config_project_dir():
-    rc = BaseConfig(private_data_dir='/tmp', project_dir='/another/path')
+def test_base_config_project_dir(tmp_path):
+    rc = BaseConfig(private_data_dir=tmp_path.as_posix(), project_dir='/another/path')
     assert rc.project_dir == '/another/path'
-    rc = BaseConfig(private_data_dir='/tmp')
-    assert rc.project_dir == '/tmp/project'
+    rc = BaseConfig(private_data_dir=tmp_path.as_posix())
+    assert rc.project_dir == tmp_path.joinpath('project').as_posix()
 
 
 def test_prepare_environment_vars_only_strings(mocker):
-    rc = BaseConfig(private_data_dir="/tmp", envvars=dict(D='D'))
+    rc = BaseConfig(envvars=dict(D='D'))
```

Comment:
In these cases, the patch will let it fall back to using `/tmp/.ansible-runner-*`, which is quite a lot better.

Reply:
I think it would be better to use `tmp_path` rather than the default temp dir of `/tmp/.ansible-runner-*`.

Reply:
What's going to happen when you start down that path is that you find a _lot_ more tests doing the initialization without specifying private_data_dir. There were _relatively_ few tests hard-coding the `'/tmp'` folder into this. The practice was also relatively specific to this test module.

So I looked back in my history from when I ran the full test suite, and did an `ls /tmp`. I paste that into a text editor. I find 1755 occurrences of such directories. These cases are absolutely _not_ specific to this test module. If I went and passed in the `tmp_path` here, I believe that would be a bad idea, because I would only cover a few cases, and would further make the test patterns _less_ consistent.

If you want to get rid of the `.ansible-runner` directories, I would suggest changing the line that does `self.private_data_dir = tempfile.mkdtemp(prefix=".ansible-runner-")` to instead get that string from a constant. Then add an `autouse` fixture that mocks the constant to somewhere in the pytest tmp paths.

Reply:
That's what I figured was happening. The correct solution is to create a fixture to mock the default path or _always_ specify `private_data_dir` for tests, as you suggested.

I would rather avoid mocking, especially since the code already accepts a parameter. That does leave the responsibility on the test author/reviewer to make sure it is always set correctly, which is a downside.

Reply:
Been thinking this over more. I think that just using the default behavior for testing is ok. Setting it for all the tests is too onerous and making a fixture plus mocking would also be a bit much.

Sorry it took me a while to come around.